### ISSUE TITLE
Verify context of a named function expression is incorrect

### ISSUE THREAD

**Issue by dvabuzyarov** [dvabuzyarov/moq.ts#1]
In case of verifying call of a named method expression with custom argument matcher the custom argument matcher is called in wrong context.

```typescript
import {Times, It, Mock, MockBehavior} from 'moq.ts';

interface IMyService {
    findSomething(value: number): number;
    doSomething(value: IData): any;
}

class MyClass {
    constructor(private myService: IMyService) {
    }

    public method(value: number): void {
        const field1 = this.myService.findSomething(value);
        this.myService.doSomething({
            field1: field1
        });
    }
}

interface IData {
    field1: number;
}

describe('linkedAccountUpdateProtoToDomainEpic', () => {
    fit('my test should be failed', () => {
        const value = 4;
        const field1 = 2;
        const myServiceMock = new Mock<IMyService>();
        myServiceMock.setBehaviorStrategy(MockBehavior.Loose);
        myServiceMock
            .setup(instance => instance.findSomething(value))
            .returns(field1);

        const myClass = new MyClass(myServiceMock.object());
        myClass.method(value);

        myServiceMock.verify(instance => instance.doSomething(It.Is<IData>(data => {
            // data: expected {field1: 2}, actual: 4;
            return data.field1 === field1;
        })), Times.Once());
    });
});
```

## Expected behavior:
This block:
```typescript
It.Is<IData>(data => {
            // data: expected {field1: 2}, actual: 4;
            return data.field1 === field1;
        }
```
should be called only once with IData object.

## Actual behavior:
it is called twice - in first time with 4 and the second time with expected parameter.

**Comment by dvabuzyarov** [comment]
8213a6b82406aa0701b730b3e3d3b9d51feda912 Fixed

**Comment by dvabuzyarov** [comment]
closes #1
